tests/testthat/test-hooks.R

```diff
@@ -140,6 +140,23 @@ run_test("style-files",
   )
 )
 
+# verify that style-files, which uses docopt with preprocessed args,
```

Comment:
As per my last comment in the old PR, I'd like to make testing for the functionality *{precommit} supports files with spaces* independent of the hooks because otherwise, as you said, we have to test every hook separately. Hence, we only need to test that `precommit_docopt()` works correctly
1. with `docs` and always provide and one or more file arguments and no option (like `--warn-only`).
2. with `docs` and always provide and one or more file arguments and options (like `--warn-only`).

The `docs` need to contain the option definition. These tests don't run with `run_test()`, but just plain {testthat} tests. Can you add them to `tests/testthat/test-docopt.R`?. Here's the scaffold: 
```r
test_that("custom docopt interface parses as expected", {
  args_variants <- list(
    'file a.R',
    c("file a.R", "file-B.R"),
    c('File A.R', 'File c.R'),
    c('Another file with spaces.R', '--warn-only')
  )
  
  "Run lintr on R files during a precommit.
Usage:
  cmdtest [--warn_only] <files>...
Options:
  --warn_only  Placeholder for test.
" -> doc
  
  for (args in args_variants) {
    args <- precommit_docopt(docs, args)
    # TODO test that args is what we want
  }
})
```

If you really want, you can leave the styler test below as an integration test, but parsable-R test we can drop.